src/System.Management.Automation/engine/SpecialVariables.cs

```diff
@@ -321,25 +326,29 @@ internal static class SpecialVariables
                                                                    /* PSCommandPath */     typeof(string),
                                                                  };
 
-        internal static readonly string[] PreferenceVariables = {
-                                                                    SpecialVariables.DebugPreference,
-                                                                    SpecialVariables.VerbosePreference,
-                                                                    SpecialVariables.ErrorActionPreference,
-                                                                    SpecialVariables.WhatIfPreference,
-                                                                    SpecialVariables.WarningPreference,
-                                                                    SpecialVariables.InformationPreference,
-                                                                    SpecialVariables.ConfirmPreference,
-                                                                };
-
-        internal static readonly Type[] PreferenceVariableTypes = {
-                                                                    /* DebugPreference */       typeof(ActionPreference),
-                                                                    /* VerbosePreference */     typeof(ActionPreference),
-                                                                    /* ErrorPreference */       typeof(ActionPreference),
-                                                                    /* WhatIfPreference */      typeof(SwitchParameter),
-                                                                    /* WarningPreference */     typeof(ActionPreference),
-                                                                    /* InformationPreference */ typeof(ActionPreference),
-                                                                    /* ConfirmPreference */     typeof(ConfirmImpact),
-                                                                  };
+        internal static readonly string[] PreferenceVariables =
+        {
+            SpecialVariables.DebugPreference,
+            SpecialVariables.VerbosePreference,
+            SpecialVariables.ErrorActionPreference,
+            SpecialVariables.WhatIfPreference,
+            SpecialVariables.WarningPreference,
+            SpecialVariables.InformationPreference,
+            SpecialVariables.ConfirmPreference,
+            SpecialVariables.PSNativeCommandUseErrorActionPreference,
```

Comment:
This variable doesn't need to be put in `PreferenceVariables` array and the corresponding arrays below.

The reason to have this array is because the existing elements in the array are mapped to a common parameter of cmdlet/advanced function. The way for preference common parameter to work in script is to set the corresponding preference variable and restore it properly, so there would be lots of variable setting operations to those preference variables. These arrays are for optimizing the setting of the preference variable -- store those preference variables as local tuples for each scope.

As for `PSNativeCommandUseErrorActionPreference`, there is no common parameter mapped to it and thus it won't be set/restore by the engine -- it will only be changed explicitly by the user, so just a regular automation variable would be sufficient. It's not really a preference variable, it's just a variable to let native command respect `ErrorActionPreference`.

Reply:
I added a comment to document this in the code